src/seamless_communication/cli/m4t/classification_head/model.py

```diff
@@ -0,0 +1,22 @@
+from torch import nn
+
+class ClassificationHead(nn.Module):
+    def __init__(self, num_languages, num_layers):
+        super(ClassificationHead, self).__init__()
+        self.num_languages = num_languages
+        self.num_layers = num_layers
+        self.hidden_dim = None
+        self.input_dim = None
+        self.layers = None
+
+    def forward(self, x):
+        if self.layers is None:
+            self.input_dim = x.size(-1)
+            self.hidden_dim = self.input_dim
+            self.layers = nn.Sequential(
+                nn.Linear(self.input_dim, self.hidden_dim),
+                nn.ReLU(),
+                nn.Linear(self.hidden_dim, self.num_languages)
+            )
```

Comment:
I just noticed this, you're actually creating new linear layers on every forward pass. This will cause new weights to be initialized every time you run the head which means it can never be trained. This should actually be done in the constructor.
I've fixed this in my PR to this branch.

Reply:
Thanks for catching this